### DISCUSSION TITLE
Please provide the compilation process for the Windows version.

### DISCUSSION THREAD

**Discussion by tierra18g** [JvanKatwijk/qt-dab#295] [Q&A]
Hi Maintainer,

I have a need here. I need to compile a special version of Windows by myself. Some functions need to be removed. How can I compile the version for the Windows environment?
I searched all over the relevant documentation here, but I couldn't find a suitable compilation step.

How to proceed? Is there a SOP?

Thank you.

johnlai

**Comment by JvanKatwijk** [comment]
I develop under Linux and cross compile for Windows.
For cross compilation I am using mingw64-32.

with mingw32-qmake I generate from the qt-dab-x.pro file a Makefile

You have to set the configuration to your likings
CONFIG          += sdrplay-v2
CONFIG          += sdrplay-v3
CONFIG          += dabstick-linux
CONFIG          += rtl_tcp
CONFIG          += airspy
CONFIG          += hackrf
CONFIG          += lime
#CONFIG         += soapy
CONFIG          += pluto-rxtx
#CONFIG         += pluto
#CONFIG         += pluto-2
#CONFIG         += elad-device
#CONFIG         += colibri\

I assume that you will not include all devices that can be configured in
your system, comment the devices out that you are not going to use
CONFIG          += faad
#CONFIG         += fdk-aac
Select one of these two, usually faad will do

#CONFIG         += JAN
CONFIG          += preCompiled
#CONFIG         += tiiLib

The files that will be included when selecting preCompiled are nof open
source, they are private due to licensing constraints,
You should choose "tiiLib" here.
tiiLib is a wrapper around another lib, the functions are loaded in run
time, but if these functions are not available (i.e.
you do not have a tiiLib installed), no harm done, the only thing is that
the identification of the transmitters is not functioning

#very experimental, simple server for connecting to a tdc handler
#CONFIG         += datastreamer
#to handle output of embedded an IP data stream, uncomment
#CONFIG         += send_datagram

these last settings can just be ignored

I recall that a few years ago, someone reported that he had compiled under
Windows, using mingw64-32 stuff.
I am not a Windows person, I have no idea how to build an executable using
MSVC

hope this helps

PS
Have a look at the manual, there it specifies what libries are needed


Op do 27 jul 2023 om 14:05 schreef tierra18g ***@***.***>:

> Hi Maintainer,
>
> I have a need here. I need to compile a special version of Windows by
> myself. Some functions need to be removed. How can I compile the version
> for the Windows environment?
> I searched all over the relevant documentation here, but I couldn't find a
> suitable compilation step.
>
> How to proceed? Is there a SOP?
>
> Thank you.
>
> johnlai
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/295>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQDOOBQ63E4OBELFSRDXSJKRFANCNFSM6AAAAAA2Z7FM3E>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Jan van Katwijk

**Comment by JvanKatwijk** [comment]
ps

If you just need a windows executable with a specific parameter setting,
just let me know.
That is easy to do for me


Op do 27 jul 2023 om 17:51 schreef jan van katwijk ***@***.***>:

> I develop under Linux and cross compile for Windows.
> For cross compilation I am using mingw64-32.
>
> with mingw32-qmake I generate from the qt-dab-x.pro file a Makefile
>
> You have to set the configuration to your likings
> CONFIG          += sdrplay-v2
> CONFIG          += sdrplay-v3
> CONFIG          += dabstick-linux
> CONFIG          += rtl_tcp
> CONFIG          += airspy
> CONFIG          += hackrf
> CONFIG          += lime
> #CONFIG         += soapy
> CONFIG          += pluto-rxtx
> #CONFIG         += pluto
> #CONFIG         += pluto-2
> #CONFIG         += elad-device
> #CONFIG         += colibri\
>
> I assume that you will not include all devices that can be configured in
> your system, comment the devices out that you are not going to use
> CONFIG          += faad
> #CONFIG         += fdk-aac
> Select one of these two, usually faad will do
>
> #CONFIG         += JAN
> CONFIG          += preCompiled
> #CONFIG         += tiiLib
>
> The files that will be included when selecting preCompiled are nof open
> source, they are private due to licensing constraints,
> You should choose "tiiLib" here.
> tiiLib is a wrapper around another lib, the functions are loaded in run
> time, but if these functions are not available (i.e.
> you do not have a tiiLib installed), no harm done, the only thing is that
> the identification of the transmitters is not functioning
>
> #very experimental, simple server for connecting to a tdc handler
> #CONFIG         += datastreamer
> #to handle output of embedded an IP data stream, uncomment
> #CONFIG         += send_datagram
>
> these last settings can just be ignored
>
> I recall that a few years ago, someone reported that he had compiled under
> Windows, using mingw64-32 stuff.
> I am not a Windows person, I have no idea how to build an executable using
> MSVC
>
> hope this helps
>
> PS
> Have a look at the manual, there it specifies what libries are needed
>
>
> Op do 27 jul 2023 om 14:05 schreef tierra18g ***@***.***>:
>
>> Hi Maintainer,
>>
>> I have a need here. I need to compile a special version of Windows by
>> myself. Some functions need to be removed. How can I compile the version
>> for the Windows environment?
>> I searched all over the relevant documentation here, but I couldn't find
>> a suitable compilation step.
>>
>> How to proceed? Is there a SOP?
>>
>> Thank you.
>>
>> johnlai
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/JvanKatwijk/qt-dab/discussions/295>, or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACCPHQDOOBQ63E4OBELFSRDXSJKRFANCNFSM6AAAAAA2Z7FM3E>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: ***@***.***>
>>
>
>
> --
> Jan van Katwijk
>
>
>
>

-- 
Jan van Katwijk

**Reply by tierra18g** [reply]
Hi Maintainer,

Thanks in advance for your reply.

I am not very familiar with the use of qmake here? How to specify the target platform for compilation? For example, win32 and so on.
I haven't seen anything like this.

OS Version: Ubuntu V22.04

Install Package:
  apt-get install git cmake
  apt-get install libandfile1
  apt-get install libandfile1-dev
  apt-get install libfaad-dev
  apt-get install libfftw3-dev portaudio19-dev
  apt-get install libgl1-mesa-dev libqt5opengl5-dev
  apt-get install libqt5svg5*
  apt-get install libqwt-headers
  apt-get install libqwt-qt5
  apt-get install libsamplerate0-dev libqwt-qt5-dev
  apt-get install libusb-1.0-0-dev mesa-common-dev
  apt-get install mingw-w64
  apt-get install mingw-w64-tools
  apt-get install sndfile
  apt-get install sndfile-programs
  apt-get install zlib1g-dev rtl-sdr
  apt-get install qtbase5-dev
  
Code Version:
  Latest Version

Compile Procedure:

  cd qt-dab/qt-dab-s5
  qmake
  make
  

Also, I pay for my profile. Please help to confirm whether there is a wrong setting there.

https://drive.google.com/drive/folders/10B8Tf9X8_573XBt5IZ3uKMyldnumbY40?usp=sharing

Thank you.

Johnlai

**Reply by tierra18g** [reply]
Also, on Ubuntu V22.04, the package mingw32-qmake is missing.

May I ask which os version you used to compile it?

Thanks

**Reply by tierra18g** [reply]
Another question, is this mingw32-qmake a standalone package? Or is it compiled from mingw-w64?

The qmake on my side is the execution file provided by QT. Do I need to look for qmake for mingw (win32)?

Sorry, please clarify again.

Thanks.

**Comment by JvanKatwijk** [comment]
I use Fedora since that has all packahes - including yhose for Qt - for the
mingw64-32 set,
Long time ago I downloaded mingw64 in Windows and was - reasonably easy -
able to compile the Qt-DAB sources
and create an executable.

Note that for cross compilation you need to create an environment in
Windows were all required dll's are installed,
that is why I generate an installer for windows.
The installer creates a folder (directory) in C:\program files(86) that
contains all that is needed



Op vr 28 jul 2023 om 04:23 schreef tierra18g ***@***.***>:

> Also, on Ubuntu V22.04, the package mingw32-qmake is missing.
>
> May I ask which version you used to compile it?
>
> Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/295#discussioncomment-6571284>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQE3LERCTCS34ZPSKWTXSMPBRANCNFSM6AAAAAA2Z7FM3E>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Comment by JvanKatwijk** [comment]
are you making progress?
The easiest way - if you have a decent Windows environment, which is
probably the case since you want to compile for windows -
is to install msys/mingw64.
The "system" has the same structure as opensuse, with an excellent package
management system.
As far as I recall in that package management system all (or at least
almost all) required libraries was available,
including all Qt include files and libraries

Unfortunately, the computer I had that running on stopped functioning last
summer, so I cannot "replay" the process there

good luck



Op vr 28 jul 2023 om 09:13 schreef jan van katwijk ***@***.***>:

> I use Fedora since that has all packahes - including yhose for Qt - for
> the mingw64-32 set,
> Long time ago I downloaded mingw64 in Windows and was - reasonably easy -
> able to compile the Qt-DAB sources
> and create an executable.
>
> Note that for cross compilation you need to create an environment in
> Windows were all required dll's are installed,
> that is why I generate an installer for windows.
> The installer creates a folder (directory) in C:\program files(86) that
> contains all that is needed
>
>
>
> Op vr 28 jul 2023 om 04:23 schreef tierra18g ***@***.***>:
>
>> Also, on Ubuntu V22.04, the package mingw32-qmake is missing.
>>
>> May I ask which version you used to compile it?
>>
>> Thanks
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/JvanKatwijk/qt-dab/discussions/295#discussioncomment-6571284>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACCPHQE3LERCTCS34ZPSKWTXSMPBRANCNFSM6AAAAAA2Z7FM3E>
>> .
>> You are receiving this because you commented.Message ID:
>> ***@***.***>
>>
>
>
> --
> Jan van Katwijk
>
>
>
>

-- 
Jan van Katwijk

**Comment by JvanKatwijk** [comment]
On my Fedora box I download from the repository all mingw32 packages that I
need (except libsndfile and libsamplerate) for compiling Qt-DAB
Included a picture of the dnf widget with  packages that are downloaded
As you can see, they are labeled mingw32-xxx
best




Op ma 31 jul 2023 om 03:45 schreef tierra18g ***@***.***>:

> Another question, is this mingw32-qmake a standalone package? Or is it
> compiled from mingw-w64?
>
> The qmake on my side is the execution file provided by QT. Do I need to
> look for qmake for mingw (win32)?
>
> Sorry, please clarify again.
>
> Thanks.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/295#discussioncomment-6590199>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQHWWOKYCSAF2BU55HLXS4E5DANCNFSM6AAAAAA2Z7FM3E>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Reply by tierra18g** [reply]
I am in a Windows environment, install MinGW-w64, and use cmake/mingw32-make to compile.
I found that there are many places in the code that need to be modified specifically for Windows.
So is this correct? Is there a special setting to compile for Windows?

[BuildFailedPic](https://drive.google.com/file/d/1-E7-GL4vIvu0bPFzbbl6QNpKCBCouN1U/view?usp=sharing)

Thanks.

**Comment by JvanKatwijk** [comment]
cross compiling from within Fedira - using minw32 - does not need changes.
As far as I recall there  I could compile the
regular source tree in windows with the mingw32 set. If usleep is a
problem, then you should replace that by Sleep which is
the windows function, but be aware that Sleep works with milleseconds
rather than micro seconds. Does not seem a big deal to me
I see you use cmake, why not use Qmake instead



Op di 1 aug 2023 om 11:51 schreef tierra18g ***@***.***>:

> I am in a Windows environment, install MinGW-w64, and use
> cmake/mingw32-make to compile.
> I found that there are many places in the code that need to be modified
> specifically for Windows.
> So is this correct? Is there a special setting to compile for Windows?
>
> BuildFailedPic
> <https://drive.google.com/file/d/1-E7-GL4vIvu0bPFzbbl6QNpKCBCouN1U/view?usp=sharing>
>
> Thanks.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/295#discussioncomment-6603809>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQBIX3XVBXCEDWHQANLXTDGTJANCNFSM6AAAAAA2Z7FM3E>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Reply by tierra18g** [reply]
Hi Maintainer,

I use qmake to generate makefile before cmake, but get some error cannot be resolved.
Please see following picture for error simulation.

Qmake generate makefile wrong.
[Pic1](https://drive.google.com/file/d/1-G76_9nq_UepfuSmBqv-cjhsIhGLZos-/view?usp=sharing)

Makefile.debug
[Makefile.debug](https://drive.google.com/file/d/1GKqy5f9Z0p1X-VRQXpWwU1Pp5d6l-lep/view?usp=sharing
)
Mingw-w64  win10 installation.
[Pic2](https://drive.google.com/file/d/1GNmzUH1_QGho07tNxKpx19ODR26exY1w/view?usp=sharing)


Johnlai

**Comment by JvanKatwijk** [comment]
Seems a little strange that you run qmake rather than mingw32-qmake
The qmake process should produce a release version of the make file


Op wo 2 aug 2023 om 03:17 schreef tierra18g ***@***.***>:

> Hi Maintainer,
>
> I use qmake to generate makefile before cmake, but get some error cannot
> be resolved.
> Please see following picture for error simulation.
>
> Qmake generate makefile wrong.
> Pic1
> <https://drive.google.com/file/d/1-G76_9nq_UepfuSmBqv-cjhsIhGLZos-/view?usp=sharing>
>
> Makefile.debug
> Makefile.debug
> <https://drive.google.com/file/d/1GKqy5f9Z0p1X-VRQXpWwU1Pp5d6l-lep/view?usp=sharing>
> Mingw-w64 win10 installation.
> Pic2
> <https://drive.google.com/file/d/1GNmzUH1_QGho07tNxKpx19ODR26exY1w/view?usp=sharing>
>
> Johnlai
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/295#discussioncomment-6611162>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQBGCNBESYLGN7NNLZLXTGTBPANCNFSM6AAAAAA2Z7FM3E>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Comment by JvanKatwijk** [comment]
you are asking  a lot of questions, it seems polite to tell what you are
doing (yes, I know you want to compile qt-dab for windows) and why, what is
the puspose


Op wo 2 aug 2023 om 08:42 schreef jan van katwijk ***@***.***>:

> Seems a little strange that you run qmake rather than mingw32-qmake
> The qmake process should produce a release version of the make file
>
>
> Op wo 2 aug 2023 om 03:17 schreef tierra18g ***@***.***>:
>
>> Hi Maintainer,
>>
>> I use qmake to generate makefile before cmake, but get some error cannot
>> be resolved.
>> Please see following picture for error simulation.
>>
>> Qmake generate makefile wrong.
>> Pic1
>> <https://drive.google.com/file/d/1-G76_9nq_UepfuSmBqv-cjhsIhGLZos-/view?usp=sharing>
>>
>> Makefile.debug
>> Makefile.debug
>> <https://drive.google.com/file/d/1GKqy5f9Z0p1X-VRQXpWwU1Pp5d6l-lep/view?usp=sharing>
>> Mingw-w64 win10 installation.
>> Pic2
>> <https://drive.google.com/file/d/1GNmzUH1_QGho07tNxKpx19ODR26exY1w/view?usp=sharing>
>>
>> Johnlai
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/JvanKatwijk/qt-dab/discussions/295#discussioncomment-6611162>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACCPHQBGCNBESYLGN7NNLZLXTGTBPANCNFSM6AAAAAA2Z7FM3E>
>> .
>> You are receiving this because you commented.Message ID:
>> ***@***.***>
>>
>
>
> --
> Jan van Katwijk
>
>
>
>

-- 
Jan van Katwijk

**Reply by tierra18g** [reply]
Sorry for asking too many questions.

I want to develop an internal test tool here, and I need to display the content and data returned by the DAB protocol on the interface.
It is convenient for testers to perform functional verification on the tested object.

We need to remove unused parts, so there will be a need for recompilation.

That's all.

**Comment by JvanKatwijk** [comment]
well, then - depending on what more or less raw data you want to extract,
it seems lot and lots easier to look at e.g. dab-scanner sources,
the dab-scanner basicaly does the fromt end processig and displays a
content description - extracted from the first few data blocks -
i.e. names and attributes of the services that are varried.
In Qt-DAB well over 70 percent of the code is not related tp front end
processing


Op wo 2 aug 2023 om 12:54 schreef tierra18g ***@***.***>:

> Sorry for asking too many questions.
>
> I want to develop an internal test tool here, and I need to display the
> content and data returned by the DAB protocol on the interface.
> It is convenient for testers to perform functional verification on the
> tested object.
>
> We need to remove unused parts, so there will be a need for recompilation.
>
> That's all.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/295#discussioncomment-6615332>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQBCJYAIZS7G5F6VVTTXTIWXXANCNFSM6AAAAAA2Z7FM3E>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Comment by JvanKatwijk** [comment]
ps
Is this a student project or something commercial


Op wo 2 aug 2023 om 13:22 schreef jan van katwijk ***@***.***>:

> well, then - depending on what more or less raw data you want to extract,
> it seems lot and lots easier to look at e.g. dab-scanner sources,
> the dab-scanner basicaly does the fromt end processig and displays a
> content description - extracted from the first few data blocks -
> i.e. names and attributes of the services that are varried.
> In Qt-DAB well over 70 percent of the code is not related tp front end
> processing
>
>
> Op wo 2 aug 2023 om 12:54 schreef tierra18g ***@***.***>:
>
>> Sorry for asking too many questions.
>>
>> I want to develop an internal test tool here, and I need to display the
>> content and data returned by the DAB protocol on the interface.
>> It is convenient for testers to perform functional verification on the
>> tested object.
>>
>> We need to remove unused parts, so there will be a need for recompilation.
>>
>> That's all.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/JvanKatwijk/qt-dab/discussions/295#discussioncomment-6615332>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACCPHQBCJYAIZS7G5F6VVTTXTIWXXANCNFSM6AAAAAA2Z7FM3E>
>> .
>> You are receiving this because you commented.Message ID:
>> ***@***.***>
>>
>
>
> --
> Jan van Katwijk
>
>
>
>

-- 
Jan van Katwijk

**Reply by tierra18g** [reply]
I'm curious, what's the difference?

In addition, which version is your Fedora? I have tried to compile it with Fedora, and it does have mingw32-qmake,
Other OS, such as CentOS, Ubuntu did not find it.

Thank you.

**Comment by JvanKatwijk** [comment]
I am using the most recent fedora version. Recently updates to F38

Op vr 4 aug 2023 om 02:59 schreef tierra18g ***@***.***>:

> I'm curious, what's the difference?
>
> In addition, which version is your Fedora? I have tried to compile it with
> Fedora, and it does have mingw32-qmake,
> Other OS, such as CentOS, Ubuntu did not find it.
>
> Thank you.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JvanKatwijk/qt-dab/discussions/295#discussioncomment-6633024>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCPHQFE5BT6FR3OMW45CW3XTRCQJANCNFSM6AAAAAA2Z7FM3E>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jan van Katwijk

**Comment by JvanKatwijk** [comment]
you did not answer my question on what the context of your project is, is
is a student project  or an indistrial project.
Please take note of the fact that while the software is available, there
are copyright issues


Op vr 4 aug 2023 om 09:29 schreef jan van katwijk ***@***.***>:

> I am using the most recent fedora version. Recently updates to F38
>
> Op vr 4 aug 2023 om 02:59 schreef tierra18g ***@***.***>:
>
>> I'm curious, what's the difference?
>>
>> In addition, which version is your Fedora? I have tried to compile it
>> with Fedora, and it does have mingw32-qmake,
>> Other OS, such as CentOS, Ubuntu did not find it.
>>
>> Thank you.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/JvanKatwijk/qt-dab/discussions/295#discussioncomment-6633024>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACCPHQFE5BT6FR3OMW45CW3XTRCQJANCNFSM6AAAAAA2Z7FM3E>
>> .
>> You are receiving this because you commented.Message ID:
>> ***@***.***>
>>
>
>
> --
> Jan van Katwijk
>
>
>
>

-- 
Jan van Katwijk

**Comment by tierra18g** [comment]
Sorry, ask again.

When I compile, I encounter the following error. Can I stop using tiiLibrary?
If not, how should I set it?

Environment:  
1. OS : Fedora 31 
2. Using : Mingw32-qmake + make
 
![image](https://github.com/JvanKatwijk/qt-dab/assets/23053089/70642561-1624-4619-b30d-104fd67d4a0a)

Thanks.
